IPython/terminal/embed.py

```diff
@@ -46,12 +47,28 @@ def kill_embedded(self, parameter_s=''):
                    "once you exit.")
 
 
+    @line_magic
+    def exit_raise(self, parameter_s=''):
+        """%exit_raise Make the current embedded kernel exit and raise and exception.
+
+        This function (after asking for confirmation) sets an internal flag so
```

Comment:
I don't think it will ask for confirmation. Confusingly, `ask_exit()` means 'ask the shell to exit', not 'ask the user if they want to exit'.

However, if someone has typed out `exit_raise`, I think we can be pretty sure that's what they want, so I'd just change the docstring.